addons/crm/models/crm_lead.py

```diff
@@ -595,7 +595,16 @@ def write(self, vals):
         if any(field in ['active', 'stage_id'] for field in vals):
             self._handle_won_lost(vals)
 
-        write_result = super(Lead, self).write(vals)
+        # stage change between two won stages: does not chang the date_closed
+        if stage_is_won:
```

Comment:
We could instead make an realy return, making code easier to read (less indents), like
```
if not stage_is_won:
    return super().write(vals)
# split recordsets for date_closed update
...
```